### ISSUE TITLE
Docs: toolbox.networkFirst should be used if freshness headers are present

### ISSUE THREAD

**Issue by ithinkihaveacat** [GoogleChrome/sw-toolbox#136]
If resources are being served with considered cache-control headers, `toolbox.networkFirst` is probably the correct strategy to use. This probably should be called out in the documentation. (Similarly, the docs should mention that `networkFirst` (and even `networkOnly`) will not necessarily force all requests to go over the network.)

For example, if a particular resource is served with `cache-control: max-age=3600`, then with the `networkFirst` strategy, this will not actually hit the network until 3600 seconds have elapsed, since `fetch()` goes through the browser's cache. (I think the `networkFirst` is equivalent to a [`stale-while-revalidate`](https://tools.ietf.org/html/rfc5861#section-3) value of infinity.)

**Comment by jeffposnick** [comment]
Yeah, the interaction between service workers, fetch, and the browser's "normal" HTTP cache is a frequent source of confusion for developers.

Would you prefer `fetchFirst`/`fetchOnly`/`cacheThenFetch` etc.?

The existing terminology was, I believe, inspired by @jakearchibald's [bible](https://jakearchibald.com/2014/offline-cookbook/#network-falling-back-to-cache), so mentioning him in to see if he has thoughts.
